src/mongocxx/database.cpp

```diff
@@ -157,5 +164,16 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const database::impl& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
```

Comment:
Wasn't there some sort of exception throwing function that we were using in mongocxx?